### PR TITLE
service management for Node

### PR DESCRIPTION
**Pull request by mikedoan**

### CODE REVIEW

File: node/service.go [hunk 1]
```diff
@@ -0,0 +1,136 @@
...
+type ActionType byte
+
+// Constants for Action Type.
+const (
+	Start ActionType = iota
+	Stop
+)
+
+// Type is service type.
+type Type byte
```
Comment (rlan35) [inline node/service.go]: Better to be explicit "ServiceType"

File: node/service.go [hunk 2]
```diff
@@ -0,0 +1,136 @@
...
+
+// SendAction sends action to action channel which is observed by service manager.
+func (node *Node) SendAction(action *Action) {
+	node.actionChannel <- action
+}
+
+// TakeAction is how service manager handles the action.
+func (node *Node) TakeAction(action *Action) {
+	if node.serviceStore == nil {
+		utils.GetLogInstance().Error("Service store is not initialized.")
```
Comment (chaosma) [inline node/service.go]: why not start it if it's nil?

Comment (mikedoan) [reply node/service.go]: because with nil it can not get the service.

File: node/service.go (hunk 1 above)
Comment (mikedoan) [reply node/service.go]: Done

**Pull request merged**